File: python/Permutations.py
class solution:
    def backtrack(self,list,temp,nums):
        if(len(temp)==len(nums)):
            list.append(temp)
        else:
            for i in range(len(nums)):
                if(nums[i] in temp):
                    continue
                temp.append(nums[i])
                self.backtrack(list,temp,nums)
                temp.pop()
    def permutations(self,nums):
        list = [[]]
        nums.sort()
        temp = []
        self.backtrack(list,nums,temp)
        return list

nums=[1,2,3]
print(solution().permutations(nums))

File: cpp/Permutations.cpp
#include<vector>
#include<iostream>
#include<algorithm>
using namespace std;
class Solution{
    protected:
    void backtrack(vector<vector<int>> &res,vector<int> &temp, vector<int> &nums){
        if(temp.size()==nums.size()){
            res.push_back(temp);
        }else{
            for(int i = 0;i<nums.size();i++){
                if(std::find(temp.begin(),temp.end(),nums[i])!=temp.end())
                    continue;
                else{
                    temp.push_back(nums[i]);
                    backtrack(res,temp,nums);
                    temp.pop_back();
                }
            }
        }
    }
    public:
    vector<vector<int>> permute(vector<int>& nums){
        vector<vector<int>> res;
        vector<int> temp;
        backtrack(res,temp,nums);
        return res;
    }
};